synapse/storage/databases/main/roommember.py

```diff
@@ -793,6 +793,21 @@ async def is_host_joined(self, room_id: str, host: str) -> bool:
     async def is_host_invited(self, room_id: str, host: str) -> bool:
         return await self._check_host_room_membership(room_id, host, Membership.INVITE)
 
+    async def is_locally_forgotten_room(self, room_id: str) -> bool:
+        sql = """
+            SELECT 1 FROM room_memberships
+            WHERE room_id = ?
+                AND user_id LIKE ?
+                AND forgotten != 1
+            LIMIT 1
```

Comment:
```suggestion
            SELECT count(*) FROM local_current_membership
            INNER JOIN room_memberships USING (room_id, event_id)
            WHERE 
                room_id = ?
                AND forgotten = 0;
```

I think ^ is an easier query, and returns the number of users who haven't forgotten the room. This avoids the `LIKE` heuristic, and handles the case where a user leaves a room, forgets it, then rejoins (as `room_memberships` is a table of *all* membership events that have happened, not just the current ones).

Reply:
I think that case is already handled, because it would add a new entry to room_memberships that has forgotten = 0, so my query would return something. But I think this is still good because it avoids the LIKE clause

Reply:
Oh yes, very true.